### PR TITLE
Use `system_command` instead of `Utils.popen_read`.

### PR DESCRIPTION
**Pull request by reitermarkus**
Previously, output would be broken when using `HOMEBREW_DEBUG=1` and `--json`, since `odebug` outputs to `stdout` (maybe it should always output to `stderr`?). 

`SystemCommand` already automatically handles `HOMEBREW_DEBUG`, so let's use that instead of manually calling `odebug` for every command here.

Bug seen here: https://github.com/reitermarkus/dotfiles/actions/runs/6435999795/job/17478462126#step:3:14451

### CODE REVIEW

File: lib/service/formula_wrapper.rb [hunk 1]
```diff
@@ -207,24 +209,23 @@ def load_service
...
       @status_output_success_type ||= if System.launchctl?
-        cmd = [System.launchctl.to_s, "list", service_name]
-        output = Utils.popen_read(*cmd).chomp
-        if $CHILD_STATUS.present? && $CHILD_STATUS.success? && output.present?
+        result = system_command(System.launchctl.to_s, args: ["list", service_name])
+        output = result.stdout.chomp
+
+        if result.success? && output.present?
           success = true
-          odebug cmd.join(" "), output
```
Comment (MikeMcQuaid) [inline lib/service/formula_wrapper.rb:214]: Does this still output the actual output of these commands or just the command being run?

Comment (reitermarkus) [reply lib/service/formula_wrapper.rb:214]: Only the command being run.

Comment (SMillerDev) [reply lib/service/formula_wrapper.rb]: The output is the important part though

Comment (reitermarkus) [reply lib/service/formula_wrapper.rb]: Do we need this debug output particularly often that it warrants outputting more than just the command? I think this is the only instance where we use `debug` this way.

Comment (SMillerDev) [reply lib/service/formula_wrapper.rb]: The command is actually the least relevant part of this function. The output will help developers figure out why something doesn't show up in the output.

Comment (MikeMcQuaid) [reply lib/service/formula_wrapper.rb]: > Do we need this debug output particularly often that it warrants outputting more than just the command?

Yes. I've done a lot of development on `brew services` recently and this has been extremely useful.

If you're not doing `brew services` development: I have no idea why you're running this with `HOMEBREW_DEBUG` or `--debug`?

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]
Needs to output the command output.

File: lib/service/formula_wrapper.rb (hunk 1 above)
Comment (reitermarkus) [reply lib/service/formula_wrapper.rb]: It's not that I'm running `brew services` specifically in debug mode, I'm running everything in debug mode in CI.

Comment (MikeMcQuaid) [reply lib/service/formula_wrapper.rb]: Yeh, don't do that for your dot files 😁

**Pull request closed without merging**